presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergCommonModule.java

```diff
@@ -200,6 +202,21 @@ public StatisticsFileCache createStatisticsFileCache(IcebergConfig config, MBean
         return new StatisticsFileCache(delegate);
     }
 
+    @Singleton
+    @Provides
+    public ManifestFileCache createManifestFileCache(IcebergConfig config, MBeanExporter exporter)
+    {
+        Cache<ManifestFileCacheKey, ManifestFileCachedContent> delegate = CacheBuilder.newBuilder()
```

Comment:
Should we just use Caffeine as the caching library since iceberg-core already brings it in ? It appears to have [better performance](https://github.com/ben-manes/caffeine/wiki/Benchmarks) and is [recommended by the Guava team](https://guava.dev/releases/snapshot/api/docs/com/google/common/cache/package-summary.html) too

Reply:
I had the same thought too. However, most of the Presto codebase uses guava caches. Since caffeine and guava are different types, it would not be compatible with the current infrastructure such as the `CacheStatsMBean` object. Additionally, the way caches are implemented now use guava's `SimpleForwardingCache` which is not available in caffeine, so I would have to roll my own. Not a terrible amount of effort, but I think there's enough work there to push that effort into a separate PR